### ISSUE TITLE
Default ID column's dynamic default can be altered

### ISSUE THREAD

**Issue by mathemancer** [centerofci/mathesar#956]
## Description

As noted in [this comment](https://github.com/centerofci/mathesar/pull/941#pullrequestreview-844879976) by @kgodey on #941 , it's currently possible to alter default columns via the API through `PATCH` requests.

## Expected behavior

It should be impossible to change the default generator of the default `id` column via the API.  In particular, if the user alters the default value, they'll have trouble since the column is required to remain unique, and they probably won't want to pick an unused integer for every row.

## To Reproduce

Give a `PATCH` request that to the appropriate columns endpoint that alters the default value of the `id` column.  The form of this request is currently in flux, so we will leave specifying it for a bit later.

## Additional context

Should it be possible to alter other attributes of the default `id` column, e.g., the name?  @kgodey @ghislaineguerin I'd appreciate a quick weigh-in on that.

**Comment by kgodey** [comment]
@mathemancer I don't think this should be specific to the `id` column, we should not allow static defaults for any column with a unique or primary key constraint.

I don't think we need any specific behavior for the `id` column, it's fine if users change the name or other attributes.

**Comment by mathemancer** [comment]
Okay, I'll alter this issue accordingly.

**Comment by Aritra8438** [comment]
Hi, @mathemancer, @kgodey, it is currently possible to alter dynamic default value settings via the API through ```PATCH``` requests, **even for primary key column, where dynamic default is necessary**.

And, after adding a record into the table, the error says ```The requested insert violates a uniqueness constraint```, therefore breaking the insertion functinality.

![image](https://user-images.githubusercontent.com/64671908/229691508-cdcffa9e-e47d-4e23-9936-c9e4c100f131.png)

So, our task is to
- **Restrict the PATCH request** such that it can't alter the dynamic default of a column? **OR**
- **Restrict the PATCH request** such that it can't alter the dynamic default of a column **which has a uniqueness constraint**?

Also, can I be assigned to this issue?

**Comment by seancolsen** [comment]
Unassigning due to inactivity

**Comment by mathemancer** [comment]
Added `needs: troubleshooting` since I can't currently reproduce it with the method in the description, but I also don't remember ever solving this bug. Thus, we need to investigate to determine whether this is still occurring or not.

**Comment by zackkrida** [comment]
I also can't reproduce this. I am going to close it, even on the chance it is still relevant.
